source/particle/property/crystal_preferred_orientation.cc

```diff
@@ -677,10 +682,65 @@ namespace aspect
 
         // create shortcuts
         const std::array<double, 4> &tau = ref_resolved_shear_stress;
-
         std::vector<double> strain_energy(n_grains);
         double mean_strain_energy = 0;
 
+        // CPX and olivine_d_fabric has different slip systems
+        // first initiate the l and n for olivine A,B,C,E types
+        std::array<Tensor<1,3>,4> slip_normal_reference {{Tensor<1,3>({0,1,0}),Tensor<1,3>({0,0,1}),Tensor<1,3>({0,1,0}),Tensor<1,3>({1,0,0})}};
+        std::array<Tensor<1,3>,4> slip_direction_reference {{Tensor<1,3>({1,0,0}),Tensor<1,3>({1,0,0}),Tensor<1,3>({0,0,1}),Tensor<1,3>({0,0,1})}};
+        if (deformation_type == DeformationType::clinopyroxene)
+          {
+            // more accurate way to calculate slip plane <110> normal by doing cross product
+            // First CPX crystal structure info, length unit Angstrom (Å)
+            //see Clinopyroxene on mindat.org at https://www.mindat.org/min-7630.html
+            const Tensor<1,3> vec_a_axis ({9.658, 0., 0.});
+            //const Tensor<1,3> vec_a_axis ({1.658, 0., 0.});
+            const Tensor<1,3> vec_b_axis ({0.,8.795, 0.});
+            const double length_c = 5.294; //[Angstrom]
+            const double angle_beta = 107.42 / 180.* M_PI; //107.42 degrees turns into rad
+            const Tensor<1,3> vec_c_axis ({length_c * std::cos(angle_beta), 0, length_c * std::sin(angle_beta)});
+
+            // tensor 1 is vector b axis minus the vector a axis
+            // Calculate plane110_normal --> crystal axis b minus axis a
+            Tensor<1,3> vec_b_minus_a = vec_b_axis - vec_a_axis;
```

Comment:
make all variables that you do not change afterwards `const`, i.e. here:
```suggestion
            const Tensor<1,3> vec_b_minus_a = vec_b_axis - vec_a_axis;
```